extensions/iam/decentralized-identity/identity-did-crypto/src/test/java/org/eclipse/dataspaceconnector/iam/did/crypto/credentials/VerifiableCredentialFactoryTest.java

```diff
@@ -9,78 +9,154 @@
+        assertThat(vc.getJWTClaimsSet().getAudience()).containsExactly("test-audience");
+        assertThat(vc.getJWTClaimsSet().getJWTID()).satisfies(UUID::fromString);
         assertThat(vc.getJWTClaimsSet().getExpirationTime()).isEqualTo(now.plus(10, MINUTES).truncatedTo(SECONDS));
     }
 
     @Test
-    void ensureSerialization() throws ParseException {
-        var vc = VerifiableCredentialFactory.create(privateKey, Map.of("did-url", "someUrl"), "test-connector", clock);
+    void ensureSerialization() throws Exception {
+        var vc = VerifiableCredentialFactory.create(privateKey, "test-connector", "test-audience", clock);
 
         assertThat(vc).isNotNull();
         String jwtString = vc.serialize();
 
         //deserialize
-        var deserialized = VerifiableCredentialFactory.parse(jwtString);
+        var deserialized = SignedJWT.parse(jwtString);
 
         assertThat(deserialized.getJWTClaimsSet()).isEqualTo(vc.getJWTClaimsSet());
         assertThat(deserialized.getHeader().getAlgorithm()).isEqualTo(vc.getHeader().getAlgorithm());
         assertThat(deserialized.getPayload().toString()).isEqualTo(vc.getPayload().toString());
     }
 
     @Test
-    void verifyJwt() throws JOSEException {
-        var vc = VerifiableCredentialFactory.create(privateKey, Map.of("did-url", "someUrl"), "test-connector", clock);
-        String jwtString = vc.serialize();
+    void verifyJwt_OnInvalidSignature_fails() {
+        var jwt = VerifiableCredentialFactory.create(privateKey, "test-connector", "test-audience", clock);
+        var unrelatedPublicKey = new EcPublicKeyWrapper(generateKeyPairP256());
+        assertThat(VerifiableCredentialFactory.verify(jwt, unrelatedPublicKey, "test-audience").getFailureMessages()).containsExactly("Invalid signature");
+    }
 
-        //deserialize
-        var jwt = VerifiableCredentialFactory.parse(jwtString);
-        var pubKey = TestHelper.readFile("public_p256.pem");
+    @Test
+    void verifyJwt_OnVerificationFailure_fails() throws Exception {
+        var jwt = mock(SignedJWT.class);
+        var message = FAKER.lorem().sentence();
+        when(jwt.verify(any())).thenThrow(new JOSEException(message));
+        assertThat(VerifiableCredentialFactory.verify(jwt, publicKey, "test-audience").getFailureMessages()).containsExactly(message);
+    }
 
-        assertThat(VerifiableCredentialFactory.verify(jwt, (ECKey) ECKey.parseFromPEMEncodedObjects(pubKey))).isTrue();
+    @Test
+    void verifyJwt_OnInvalidClaims_fails() throws Exception {
+        var jwt = mock(SignedJWT.class);
+        var message = FAKER.lorem().sentence();
+        when(jwt.verify(any())).thenReturn(true);
+        when(jwt.getJWTClaimsSet()).thenThrow(new ParseException(message, 0));
+        assertThat(VerifiableCredentialFactory.verify(jwt, publicKey, "test-audience").getFailureMessages()).containsExactly(message);
+    }
+
+    @ParameterizedTest(name = "{2}")
+    @MethodSource("claimsArgs")
```

Comment:
`@ArgumentsSource` can be used, and gives advantages like type check, no need for the argument source class to be public

Reply:
Could you clarify this? Examples I see of using `@ArgumentsSource` in EDC use the same method as here, just wrapped in a static class. I've removed the `public` qualifier on the method, as it was indeed not needed